lib/date_utils.js

```diff
@@ -88,12 +88,37 @@ function createEdgeDate(edge, unit) {
     return date;
 }
 
+function createDatePiece(year, month, day, time) {
+    // All non-supplied values to the left of the largest supplied
+    // value are set to the present. All non-supplied values to the
+    // right of the largest supplied value are set to the minimum.
+    const date = new Date;
+    if (year > 0) {
+        date.setYear(year);
+        date.setMonth(0, 1); // 1st of Jan
+        date.setHours(0, 0, 0, 0);
+    }
+    if (month > 0) {
+        date.setMonth(month - 1, 1);
+        date.setHours(0, 0, 0, 0);
+    }
+    if (day > 0) {
+        date.setDate(day);
+        date.setHours(0, 0, 0, 0);
+    }
+    if (time !== null)
+        date.setHours(time.hour, time.minute, 0, 0);
```

Comment:
time.seconds too. And also, time.seconds can be fractional to have millisecond resolution, so don't pass the last `0` to setHours.